### PR TITLE
Speed up add cell rates

### PR DESCRIPTION
**Pull request by totto82**
PR #1555 caused a almost 5% regression of the run time of most models. The reason is that now the update of the intensive quantities is done in the well model in the method updatePerforationIntensiveQuantities() which are not OpenMP parallel . This PR makes sure that only the intensive quantities of the perforated cells are updated and thus restores the performance.

NB. There are several other call to  updatePerforationIntensiveQuantities() in the well model that are not  OpenMP parallel, that could cause similar unexpected problems in the future. These should be addressed in later PRs. But currently they do not cause any performance regression, since they are only called when the intensive quantities are cached.

### CODE REVIEW

**Comment by totto82** [comment]
jenkins build this please

File: opm/autodiff/WellInterface.hpp [hunk 1]
```diff
@@ -195,8 +195,16 @@ namespace Opm
         virtual void addWellContributions(Mat&) const
         {}
 
-        virtual void addCellRates(RateVector& rates, int cellIdx) const
-        {}
+        void addCellRates(RateVector& rates, int cellIdx) const
```
Comment (GitPaean) [inline opm/autodiff/WellInterface.hpp]: tiny nittpicking, please move this to the `_impl.hpp` file.

**Comment by totto82** [comment]
jenkins build this please

File: opm/autodiff/BlackoilWellModel_impl.hpp [hunk 2]
```diff
@@ -228,6 +230,10 @@ namespace Opm {
             well->init(&phase_usage_, depth_, gravity_, number_of_cells_);
         }
 
+        for (auto& well : well_container_) {
```
[the remark this answers is not in the document]
Comment (totto82) [reply opm/autodiff/BlackoilWellModel_impl.hpp]: Fixed.

File: opm/autodiff/WellInterface.hpp (hunk 1 above)
Comment (atgeirr) [reply opm/autodiff/WellInterface.hpp]: I agree, we should follow the existing pattern here (although I see now that was broken recently in the ebos API merge, and I did not notice).

**Comment by atgeirr** [comment]
Looks ready now, merging!

File: opm/autodiff/WellInterface_impl.hpp [hunk 3]
```diff
@@ -1125,5 +1137,18 @@ namespace Opm
...
 
+    template<typename TypeTag>
+    void
+    WellInterface<TypeTag>::addCellRates(RateVector& rates, int cellIdx) const
+    {
+        for (int perfIdx = 0; perfIdx < number_of_perforations_; ++perfIdx) {
+            if (cells()[perfIdx] == cellIdx) {
+                for (int i = 0; i < RateVector::dimension; ++i) {
+                    rates[i] += connectionRates_[perfIdx][i];
+                }
```
Comment (GitPaean) [inline opm/autodiff/WellInterface_impl.hpp]: maybe there should be a `break;` within the `if` here?

**Pull request merged**